### DISCUSSION TITLE
Allow sound effects to applied per sound/audio node

### DISCUSSION THREAD

**Discussion by yosimba2000** [godotengine/godot-proposals#8190] [Engine Core]
Right now, audio effects only work on the audio Buses and cannot be applied to the audio nodes/streams themselves.

For example, I want to apply both a reverb and low-pass filter to a specific audio stream. The only way to achieve this is to create a new audio bus and apply the effects to the bus, then route the audio to the new bus.

If I then have a 2nd audio stream that I want to apply different values of the same effects as the first audio bus, I cannot simply route my 2nd stream into the 1st bus because adjusting the values of the 1st bus would also change the output of the 1st stream. So therefore, I would need to create a 2nd audio bus specifically to apply effects to the 2nd audio stream. Therefore, Buses cannot usually be shared by multiple audio streams because then each stream would not be able to have separate effects. 

So as you can see, buses are best used as PRESETS. But in a game where each audiostream should be able to be modified independently from all other streams, that means each audiostream needs its own unique bus because they cannot share buses with other streams. This doesn't seem very efficient, and would be resolved by allowing us to apply effects directly to audio nodes/streams themselves.

**Comment by RoyBerardo** [comment]
It seems that it didn't auto-link, so I'm going to put this here for anyone who comes by this - There is now a PR for this proposal:
https://github.com/godotengine/godot/pull/107523

**Reply by yosimba2000** [reply]
This is amazing, thank you so much!!
